providers/edge3/docs/architecture.rst

```diff
@@ -0,0 +1,189 @@
+ .. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Edge Provider Architecture
+==========================
+
+Airflow consist of several components which are connected like in the following diagram. The Edge Worker which is
+deployed outside of the central Airflow cluster is connected via HTTP(s) to the API server of the Airflow cluster:
+
+.. graphviz::
```

Comment:
How do we maintain the images created here?
I assume we need to remember manually run this code and upload the image manually?
In other cases we used diagram lib and automatic update proccess with pre-commit https://github.com/apache/airflow/pull/46056/files would that be applicable here?

Reply:
Yes we can... would propose this as a future improvement. Until then I'd manually update.

Reply:
Note that I started with this as a replica of the one in Celery in https://github.com/apache/airflow/blob/main/providers/celery/docs/celery_executor.rst?plain=1#L103

Reply:
@pykenny maybe you'd be interested in giving it a try? As you already handled #46056

Reply:
@eladkal 

I think we need to start with modifying pre commit hooks:

https://github.com/apache/airflow/blob/c86a4e1582fde083a4c93ea81955f586009177d1/.pre-commit-config.yaml#L486-L495

Currently, pre-commit hook for updating diagrams only covers scripts under these folders:
 - Python scripts with `diagram_` prefix  under `airflow-core/docs/[SOME_SECTION]/` folders
 - Python scripts under `docs/images/` folder

While provider documentations are now placed within folder that looks like `providers/[PROVIDER_TITLE]/docs/`, which is not covered by the path pattern set up in the hook.

Reply:
And... do we have (documented) convention for where to place a kinds of assets? From existing rule Airflow's having both `images` and `img` folders to place images, and that confused me a bit.

Reply:
Also some initial attempts to plot the two graphviz images with `diagrams` library:

|![image](https://github.com/user-attachments/assets/b2a44e05-602d-4acc-9fbb-d72c324d0633)|![image](https://github.com/user-attachments/assets/933e8a4a-362d-4777-9906-7012437944ba)|
|:---:|:---:|
| **Architecture** | **Worker State Model** |

Reply:
Mhm, I don't know if the new proposed images are actually "nicer" - I feel like the older ones are good as well. Do we need to change the layout engine for the new images or could we also use GraphViz which is used behind PUML as well? Then we could take-over the existing images and just use Python to attach to enums/interfaces and have a single source but generate the same results.
If mixed both "img" and "image" folders are used, I think it would be okay to normalize across repo. Might be good to have oe "normalization" PR and then adding the pre-commit.

Would it be meaningful to have a kinf of convention that the python files to generate an image would also resive in "img" folder?

Reply:
I can use `graphviz` as well since `graphviz` is required library of `diagrams`.